*** clearsilver/clearsilver.c	2009-10-28 00:05:50.000000000 +0900
--- clearsilver_for_PHP5.4.X/clearsilver.c	2013-01-22 15:46:23.060711070 +0900
***************
*** 84,90 ****
  	return err_str;
  }
  
! 
  /* {{{ clearsilver_functions[]
   *
   * Every user visible function must have an entry in clearsilver_functions[].
--- 84,92 ----
  	return err_str;
  }
  
! #if PHP_VERSION_ID >= 50400
! # define function_entry		zend_function_entry
! #endif
  /* {{{ clearsilver_functions[]
   *
   * Every user visible function must have an entry in clearsilver_functions[].
***************
*** 477,490 ****
  	}
  
  	/* start buffering */
! 	php_start_ob_buffer (NULL, 0, 1 TSRMLS_CC);
  
  	/* print zval */
  	zend_print_variable(value);
  
  	/* retrieve output buffer */
! 	php_ob_get_buffer (return_value TSRMLS_CC);
! 	php_end_ob_buffer (0, 0 TSRMLS_CC);
  	
  	err = hdf_set_value(hdf, name, Z_STRVAL_P(return_value));
  	if (err != STATUS_OK) {
--- 479,492 ----
  	}
  
  	/* start buffering */
!     php_output_start_default(TSRMLS_C);
  
  	/* print zval */
  	zend_print_variable(value);
  
  	/* retrieve output buffer */
!     php_output_get_contents(return_value TSRMLS_CC);
!     php_output_discard(TSRMLS_C);
  	
  	err = hdf_set_value(hdf, name, Z_STRVAL_P(return_value));
  	if (err != STATUS_OK) {
